src/YHttpServer.h

```diff
@@ -29,6 +29,8 @@
 #define YUI_AUTH_USER       "YUI_AUTH_USER"
 #define YUI_AUTH_PASSWD     "YUI_AUTH_PASSWD"
 
+#define YUI_API_VERSION     "v1"
```

Comment:
Better make this something that is easier to compare numerically; something like `01.00.00` (no `v` prefix). This will make life a lot easier in the future.

Reply:
For the REST API versioning paths like `/api/v1` are very common, I'd stick with that convention.

Numerical comparison does not make much sense here, you cannot know if v3 is (will be) backward compatible with v2 or not. Both server and client API versions needs to match otherwise you are asking for troubles...

Reply:
Well, okay then. I was thinking about how Qt handles this; they let you go into very deep detail what you can do depending on the exact version.